workflow-trace-viewer/src/jvmMain/kotlin/com/squareup/workflow1/traceviewer/ui/WorkflowInfoPanel.kt

```diff
@@ -103,14 +103,18 @@ private fun NodePanelDetails(
         modifier = Modifier.padding(top = 8.dp, bottom = 8.dp)
       )
     }
-    val fields = listOf("Name", "Id", "Props", "State", "Rendering")
 
-    items(fields) { field ->
-      DetailCard(
-        label = field,
-        currValue = Node.getNodeField(node.current, field),
-        pastValue = if (node.previous != null) Node.getNodeField(node.previous, field) else null
-      )
+    val fields = Node::class.memberProperties
+    for (field in fields) {
```

Comment:
oh nice. Ok i see you moved to reflection here. that's great. I think it makes sense.